pkg/commands/compute/init.go

```diff
@@ -196,25 +211,104 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
 		c.cloneFrom = from
 	}
 
-	// We only want to fetch a remote package if c.cloneFrom has been set.
-	// This can happen in two ways:
+	// There are three situations in which we might fetch something
+	// here. We might fetch a template if:
 	//
-	// 1. --from flag is set
+	// 1. --from flag is set to a template repository, or
 	// 2. user selects starter kit when prompted
 	//
+	// Or we fetch an existing, deployed package if
+	//
+	// 3. --from flag is set to a serviceID
+	//
 	// We don't fetch if the user has indicated their language of choice is
 	// "other" because this means they intend on handling the compilation of code
 	// that isn't natively supported by the platform.
 	if c.cloneFrom != "" {
-		err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
-		if err != nil {
-			c.Globals.ErrLog.AddWithContext(err, map[string]any{
-				"From":      from,
-				"Branch":    branch,
-				"Tag":       tag,
-				"Directory": c.dir,
+		if !isExistingService {
+			err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
+			if err != nil {
+				c.Globals.ErrLog.AddWithContext(err, map[string]any{
+					"From":      from,
+					"Branch":    branch,
+					"Tag":       tag,
+					"Directory": c.dir,
+				})
+				return err
+			}
+		} else {
+			profileName, _, _ := c.Globals.Profile()
+
+			var service *fastly.Service
+			var pack *fastly.Package
+			err = spinner.Process("Fetching service details", func(_ *text.SpinnerWrapper) error {
+				service, err = c.Globals.APIClient.GetService(&fastly.GetServiceInput{
+					ServiceID: c.cloneFrom,
+				})
+				if err != nil {
+					c.Globals.ErrLog.AddWithContext(err, map[string]any{
+						"From":      c.cloneFrom,
+						"Directory": c.dir,
+					})
+					if hErr, ok := err.(*fastly.HTTPError); ok && hErr.IsNotFound() {
+						return fmt.Errorf("the service %s could not be found for profile %s", c.cloneFrom, profileName)
+					}
+					return err
+				}
+
+				if service.Type == nil || *service.Type != "wasm" {
+					return fmt.Errorf("service %s is not a compute service (type is %s)", c.cloneFrom, *service.Type)
+				}
+
+				if service.ActiveVersion == nil {
+					return fmt.Errorf("service %s does not have an active version to clone", c.cloneFrom)
+				}
```

Comment:
We could return a remediation error message that lets the user know they can pass the autoclone flag:
https://github.com/fastly/cli/blob/main/pkg/errors/remediation_error.go#L106-L109

e.g.

```go
return fsterr.RemediationError{
	Inner:       fmt.Errorf("service %s does not have an active version to clone", c.cloneFrom),
	Remediation: fsterr.AutoCloneRemediation,
}
```

Then if the flag is passed, we can add the logic to make an API call to auto-clone their service for them.

Reply:
I'm not sure if cloning would help in this situation (I should use a different word in the error message). We need a version that will allow us to [fetch the package data](https://www.fastly.com/documentation/reference/api/services/package/#get-package). I tried that with a new compute service with no active versions and I get a 404.

Reply:
Oh, I see. I think I may have misread the code. Yeah you'd need at least one active version of the service for a package to be pullable from it.

Reply:
There's a slightly surprising case where:

* the service exists
* it has no active versions
* it has an _inactive_ version with a package and metadata uploaded

In order to do this you have to go into the UI, create a compute service, upload a package, then never activate or publish it. It's certainly possible, even if unlikely, and we want to make it more unlikely.

So I reworked this to try to find the newest service version that has package data associated with it. If no versions satisfy that requirement, we (currently) bail. We could create a `fastly.toml` with just the service ID in it, or fall back to prompting. My inclination is to do that as a fast-follow UX improvement, but I can certainly do it now, too.